### PR TITLE
Adding a simple persistence provider (Replacement PR)

### PR DESCRIPTION
**Pull request by erikbosch**
Replaces #11

### CODE REVIEW

**Comment by SebastianSchildt** [comment]
@MathiasDanzeisen is this functional, and  includes everything you intended to have here? Or do we need to extend this? If so, should we merge already, and add further PRs as required?

File: kuksa-persistence-provider/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+djson = { path="../../platform/modules/json-al/djson_rust/" ,  optional = true }
+
+# For testing
+[dev_dependencies]
+tempfile = "3.2.0"
+
+[features]
+default = ["json_tinyjson"]
+# use a dependable json library, this still requires tiny-json for writing
+json_djson = [ "dep:djson","dep:tinyjson"  ]
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Cargo.toml]:
```suggestion
# json_djson = [ "dep:djson","dep:tinyjson"  ] # Uncomment to use djson lib
```

File: kuksa-persistence-provider/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+log = "0.4.22"
+env_logger = "0.11.5"
+regex = "1.11.0"
+anyhow = "1.0.40"
+#Only this version is compatible with kuksa crate. 0.13 is not
+prost-types = "0.11.9"
+tokio = { version="1.40.0",  features = ["full"] }
+
+# TODO: change this to a a github path once available
+djson = { path="../../platform/modules/json-al/djson_rust/" ,  optional = true }
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Cargo.toml]:
```suggestion
# djson = { path="../../platform/modules/json-al/djson_rust/" ,  optional = true } # Uncomment to use djson lib
```

File: kuksa-persistence-provider/Readme.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    "Vehicle.VehicleIdentification.VehicleInteriorColor": {
+        "value": "Black"
+    }
+}
+```
+
+## Features
+
+* djson: Use a dependable json library.
+  Check out the repo with the depenable json library at the same level as the parent folder or project folder of this repo.
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Readme.md]:
```suggestion
  * Check out the repo with the depenable json library at the same level as the parent folder or project folder of this repo.
```

File: kuksa-persistence-provider/Readme.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  Enable with ```cargo build --features json_djson --no-default-features```
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Readme.md]:
```suggestion
  * Edit `Cargo.toml` and uncomment the following lines

    ```toml
    djson = { path="../../platform/modules/json-al/djson_rust/" ,  optional = true } # Uncommment to use djson

    # ... 

    json_djson = [ "dep:djson","dep:tinyjson"  ]   # Uncommment to use djson
    ```
  * Install rust nightly which is needed for djson at the moment [11/2024]
    ```bash
    rustup toolchain install nightly
    rustup default nightly
    ```
  * Enable with ```cargo build --features json_djson --no-default-features``````

File: kuksa-persistence-provider/Readme.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-databroker:latest --port 55556
+```
+
+Start kuksa sdk with:
+
+```bash
+docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-python-sdk/kuksa-client:latest grpc://127.0.0.1:55556
+# Check Vales in cmd
+getValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed
+```
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Readme.md]:
```suggestion

1) Check for empty data point after startup
   
    * In Terminal A): Start kuksa databroker with:

        ```bash
        docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-databroker:latest --port 55556
        ```

    * In Terminal B) Start kuksa sdk with:

        ```bash
        docker run -it --rm --net=host ghcr.io/eclipse-kuksa/kuksa-python-sdk/kuksa-client:latest grpc://127.0.0.1:55556
        # Check Vales in cmd
        getValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed
        ```

    * In Terminal C) Start kuksa persistency provider with:

        ```bash
        cargo run
        ```

2) Set data point which is stored by persistency provider
    * In Terminal B) :

        ```bash
        # Set Vales in cmd
        setValue Vehicle.Cabin.HVAC.Station.Row4.Passenger.FanSpeed 40
        ```
3) Restart and check if datapoint with old value is restored by persistency provider
```

File: kuksa-persistence-provider/Readme.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+* djson: Use a dependable json library.
+  Check out the repo with the depenable json library at the same level as the parent folder or project folder of this repo.
+  Enable with ```cargo build --features json_djson --no-default-features```
+
+
+## Build
+
+```bash
+cargo build --features json_djson --no-default-features
+```
```
Comment (MathiasDanzeisen) [inline kuksa-persistence-provider/Readme.md]:
```suggestion

```bash
cargo build 
# with features, see above for prerequisites
cargo build --features json_djson --no-default-features
```
```

**Review by MathiasDanzeisen — CHANGES_REQUESTED** [review]
Thanks for putting up the pull request.
I suggested some changes in the Cargo script to build the demo without the djson lib at hand and improvements for documentation

**Review by SebastianSchildt — APPROVED** [review]
Thank you for providing/fixing this.

Merging this now. Once djson location is stable we can revisit, and add the URL.

**Pull request merged**